### PR TITLE
Implements Black Friday banner updates for 2023

### PR DESCRIPTION
**Pull request by Camwyn**
Updates the template to be more like the Stellar Sale template.
Reorganizes the styles for the BF banner -
 - Removes them from `src/resources/postcss/tribe-common-admin/_main.pcss`
 - Adds a new file for the BF banner styles at `src/resources/postcss/tribe-common-admin/notices/_black-friday.pcss`
 - Sets up to do the same for the others as we change them.
 - Adds functionality to bring in outside/extra assets (like Google Fonts) in the Service Provider.
 - Imports the fonts for this year's banner.
 - Deprecates all the old Marketing functionality (which was all BF-specific anyway) as we now use the Date_Based approach.

Styling uses container queries for positioning: 
 - https://caniuse.com/css-container-queries
 - https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Container_Queries

 [ET-1890]

[ET-1890]: https://stellarwp.atlassian.net/browse/ET-1890?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by redscar — APPROVED** [review]
This looks great. It makes the process of changing the notice easy to understand.

File: src/Tribe/Admin/Notice/Service_Provider.php [hunk 1]
```diff
@@ -63,4 +64,8 @@ public function plugins_loaded() {
         // EOY Sale disabled for 2022
 		// tribe( Marketing\End_Of_Year_Sale::class );
 	}
+
```
Comment (codingmusician) [inline src/Tribe/Admin/Notice/Service_Provider.php:67]: Needs docs?

File: src/admin-views/notices/tribe-bf-general.php [hunk 2]
```diff
@@ -3,29 +3,37 @@
...
+			<p class="tribe-marketing-notice__content tribe-common-h4">
+				<?php _e( 'Upgrade and purchase new products during the Black Friday Sale.', 'tribe-common' ); ?>
+			</p>
+			<p>
+				<span class="tribe-marketing-notice__cta tribe-marketing-notice__cta-shop-now tribe-marketing-notice__cta-shop-now--desktop">
+					<a target="_blank" href="<?php echo esc_url( $cta_url ); ?>">
+						<?php echo esc_html_x( 'Shop now', 'Shop now link text', 'tribe-common' ) ?>
+					</a>
+				</span>
+			</p>
```
Comment (codingmusician) [inline src/admin-views/notices/tribe-bf-general.php:28]: Indents
```suggestion
	<div class="tribe-marketing-notice__content-wrapper">
		<h3 class="tribe-marketing-notice__header tribe-common-h3">
			<?php _e( 'Save 40% on The Events Calendar', 'tribe-common' ); ?>
		</h3>
		<p class="tribe-marketing-notice__content tribe-common-h4">
			<?php _e( 'Upgrade and purchase new products during the Black Friday Sale.', 'tribe-common' ); ?>
		</p>
		<p>
			<span class="tribe-marketing-notice__cta tribe-marketing-notice__cta-shop-now tribe-marketing-notice__cta-shop-now--desktop">
				<a target="_blank" href="<?php echo esc_url( $cta_url ); ?>">
					<?php echo esc_html_x( 'Shop now', 'Shop now link text', 'tribe-common' ) ?>
				</a>
			</span>
		</p>
```

File: src/Tribe/Admin/Notice/Marketing/Black_Friday.php [hunk 3]
```diff
@@ -69,4 +75,17 @@ public function get_start_time() {
...
+	 * Enqueue additional assets for the notice.
+	 *
+	 * @since 5.1.10
+	 */
+	public function enqueue_additional_assets() {
+		if ( ! $this->should_display() ) {
+			return;
+		}
+		// Adds the Montserrat font from Google Fonts.
+		wp_enqueue_style( 'tec_black_friday_font', 'https://fonts.googleapis.com/css2?family=Montserrat:wght@400;700' );
```
Comment (bordoni) [inline src/Tribe/Admin/Notice/Marketing/Black_Friday.php:89]: Why are we not using `tribe_asset`?

File: src/Tribe/Admin/Notice/Service_Provider.php (hunk 1 above)
Comment (Camwyn) [reply src/Tribe/Admin/Notice/Service_Provider.php:67]: Addressed

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Admin/Notice/Marketing.php [hunk 4]
```diff
@@ -29,7 +29,7 @@ class Tribe__Admin__Notice__Marketing {
 	public $et_is_active;
 
 	public function __construct() {
-		$tribe_dependency    = Tribe__Dependency::instance();
```
Comment (Camwyn) [inline src/Tribe/Admin/Notice/Marketing.php:32]: FYI `Tribe__Dependency::instance()` is deprecated - we should not be using it

File: src/Tribe/Admin/Notice/Marketing/Black_Friday.php (hunk 3 above)
Comment (Camwyn) [reply src/Tribe/Admin/Notice/Marketing/Black_Friday.php]: We are now 😉

**Review by borkweb — COMMENTED** [review]
I think this looks great. Just double-checking: is the end date accurate?

**Comment by Camwyn** [comment]
> I think this looks great. Just double-checking: is the end date accurate?

@borkweb Yes, those are the dates I was given. You can check the ticket as well.

**Review by borkweb — APPROVED** [review]

**Pull request merged**